lib/compilers/hlsl.js

```diff
@@ -35,17 +42,79 @@ export class HLSLCompiler extends BaseCompiler {
         super(info, env);
 
         this.compiler.supportsIntel = false;
+
+        if (default_dxc === null && this.compiler.group === 'dxc') {
+            default_dxc = this.compiler;
+        }
+
+        if (this.rga) {
+            // RGA is invoked in two steps. First, DXC is invoked to compile
+            // the SPIR-V output of the HLSL file. Next, RGA is invoked to
+            // consume the SPIR-V output and produce the requested ISA. To
+            // accomplish this two-pass compilation, a helper script rga.js
+            // is used to wrap the compilation as a single executable script.
+            this.rgaOptions = [
+                path.join(path.dirname(url.fileURLToPath(import.meta.url)), 'rga.js'),
+                this.compiler.exe,
+            ];
+
+            // Replace the RGA executable with the node path
+            this.compiler.exe = process.execPath;
+        }
+    }
+
+    // Returns true if the compiler selected is a Radeon GPU Analyzer
+    get rga() {
+        return this.compiler.group === 'rga';
+    }
+
+    // Returns true if the compiler selected is a DirectX Shader Compiler
+    get dxc() {
+        return this.compiler.group === 'dxc';
     }
 
     /* eslint-disable no-unused-vars */
-    optionsForFilter(filters, outputFilename) {
+    optionsForFilter(filters, outputFilename, userOptions) {
+        // If compiling with RGA, inject the RGA script, RGA executable,
+        // DXC executable, and output filename as the initial arguments
+        // (to be passed to node). Changes here require changes in the
+        // rga.js script.
+        if (this.rga) {
+            return [...this.rgaOptions, default_dxc.exe, outputFilename, '-Zi', '-Qembed_debug'];
```

Comment:
Here if we were to go with a separate `rga.js` we would have a configuration option to point at the `dxc` executable to use. We can even set it differently per compiler (or by group; but that's a configuration choice, not baked into code).